C/primitive/elements/exec.c

```diff
@@ -84,9 +82,15 @@ extern bool elements_simplicity_execSimplicity( simplicity_err* error, unsigned
       }
     }
     if (IS_OK(*error)) {
-      *error = fillWitnessData(dag, type_dag, (size_t)dag_len, witness);
-    }
+      bitstream witness_stream = initializeBitstream(witness, witness_len);
+      *error = fillWitnessData(dag, type_dag, (size_t)dag_len, &witness_stream);
       if (IS_OK(*error)) {
+        *error = closeBitstream(&witness_stream);
+        if (SIMPLICITY_ERR_BITSTREAM_TRAILING_BYTES == *error) *error = SIMPLICITY_ERR_WITNESS_TRAILING_BYTES;
+        if (SIMPLICITY_ERR_BITSTREAM_ILLEGAL_PADDING == *error) *error = SIMPLICITY_ERR_WITNESS_ILLEGAL_PADDING;
+      }
+    }
+    if (IS_OK(*error)) {
```

Comment:
4c77c31217a590cc933c0e27570704ec59c45cf1 If the witness is now a bitstream instead of a bitstring, then we could drop the specialized witness error variants in favor of the bitstream variants.

Reply:
I've thought about this, but I think it is more helpful to get an error message about which stack item had the wrong padding.

Reply:
Ok, in this case let's keep both variants. We could rename `SIMPLICITY_ERR_BITSTREAM_*` to `SIMPLICITY_ERR_PROGRAM_*` to clarify the distinction. If we do that, let's do that in a follow-up PR.